lib/api/utilities/functions.php

```diff
@@ -81,172 +81,173 @@ function beans_remove_dir( $dir_path ) {
+		$root_path = wp_normalize_path( untrailingslashit( ABSPATH ) );
+		$root_url  = untrailingslashit( site_url() );
 
 		// Remove subfolder if necessary.
-		if ( '' !== ( $subfolder = parse_url( $host, PHP_URL_PATH ) ) ) {
-
-			$root = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $root );
-			$host = preg_replace( '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#', '', $host );
+		$subfolder = parse_url( $root_url, PHP_URL_PATH );
+		if ( $subfolder && '/' !== $subfolder ) {
+			$pattern   = '#' . untrailingslashit( preg_quote( $subfolder ) ) . '$#';
+			$root_path = preg_replace( $pattern, '', $root_path );
+			$root_url  = preg_replace( $pattern, '', $root_url );
+		}
 
-			// Add the blog path for multsites.
-			if ( ! is_main_site() && ( $blogdetails = get_blog_details( get_current_blog_id() ) ) ) {
-				if ( ! ( defined( 'WP_SITEURL' ) ) || ( defined( 'WP_SITEURL' ) && WP_SITEURL == site_url() ) ) {
-					$host = untrailingslashit( $host ) . $blogdetails->path;
-				}
+		// If it's a multisite and not the main site, then add the site's path.
+		if ( ! is_main_site() ) {
+			$blogdetails = get_blog_details( get_current_blog_id() );
+			if ( $blogdetails && ( ! defined( 'WP_SITEURL' ) || ( defined( 'WP_SITEURL' ) && WP_SITEURL === site_url() ) ) ) {
+				$root_url = untrailingslashit( $root_url ) . $blogdetails->path;
 			}
 		}
 
-		$explode = beans_get( 0, explode( '/' , trailingslashit( ltrim( $subfolder, '/' ) ) ) );
-
 		// Maybe re-add tilde from host.
-		if ( false !== stripos( $explode, '~' ) ) {
-			$host = trailingslashit( $host ) . $explode;
+		$maybe_tilde = beans_get( 0, explode( '/', trailingslashit( ltrim( $subfolder, '/' ) ) ) );
+		if ( false !== stripos( $maybe_tilde, '~' ) ) {
+			$root_url = trailingslashit( $root_url ) . $maybe_tilde;
 		}
 	}
 
 	// Remove root if necessary.
-	if ( false !== stripos( $path, $root ) ) {
-		$path = str_replace( $root, '', $path );
+	if ( false !== stripos( $path, $root_path ) ) {
+		$path = str_replace( $root_path, '', $path );
 	} elseif ( false !== stripos( $path, beans_get( 'DOCUMENT_ROOT', $_SERVER ) ) ) {
 		$path = str_replace( beans_get( 'DOCUMENT_ROOT', $_SERVER ), '', $path );
 	}
 
-	return trailingslashit( $host ) . ltrim( $path, '/' );
-
+	return trailingslashit( $root_url ) . ltrim( $path, '/' );
 }
 
 /**
  * Convert internal url to a path.
  *
  * This function must only be used with internal urls.
  *
- * @since 1.0.0
+ * @since 1.5.0
  *
- * @param string $url Url to be converted. Accepts only internal urls.
+ * @param string $url           Url to be converted. Accepts only internal urls.
+ * @param bool   $force_rebuild (Optional) Forces the rebuild of the root url and path.
  *
  * @return string Absolute path.
  */
-function beans_url_to_path( $url ) {
-
-	static $root, $blogdetails;
+function beans_url_to_path( $url, $force_rebuild = false ) {
+	static $root_path, $blogdetails;
+	$site_url = site_url();
```

Comment:
We hit `site_url()` a couple of times.  That function has a lot going on.  Therefore, let's cache it here and then reuse the variable for a tiny little optimization bump.